### DISCUSSION TITLE
How to merge .speclib spectral libraries (DIA-NN 1.8.2 beta 27) ?

### DISCUSSION THREAD

**Discussion by Cllozano** [vdemichev/DiaNN#937] [Q&A] [answered]
Hi Vadim, 

I have made a spectral library (SL) with the DIA-NN 1.8.2 beta 27 version. In my output library folder, I only have a predicted.speclib library (and not the .tsv one). 

I noted that the -- lib command permits loading several libraries, but is restricted to .tsv format. Is there a way to run (or merge) several .speclib SL ?

Thanks
Clem

**Comment by vdemichev** [comment, marked as answer]
Hi Clem,

Sorry for the late reply.
Predicted library generation only produced .predicted.speclib, as a .tsv will often be of an unreasonable size (~10x of .speclib).

You can convert any .speclib to .tsv buy just specifying .speclib as input, .tsv as output and not specifying any raw files.

Best,
Vadim

**Comment by cutleraging** [comment]
Hi Vadim,

How can you do this in v2.1? I have a .predicted.speclib file and want to have a .tsv of this? What type of file is the .predicted.speclib?

Thanks,
Ronnie

**Reply by vdemichev** [reply]
Hi Ronnie,

If it's a .predicted.speclib created by 2.0-series, you can only convert it to .parquet (specify .speclib is input, select 'Generate spectral library', do no specify any raw files). You can work with .parquet in a number of ways (please see https://github.com/vdemichev/DiaNN?tab=readme-ov-file#editing-spectral-libraries), including conversion to .tsv.

Best,
Vadim

**Reply by cutleraging** [reply]
Thank you!

**Reply by cutleraging** [reply]
The RT and the IM in the speclib, what units are these in?
